packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1808,12 +1807,12 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     _internalScrollController?.dispose();
     _currentAutofillScope?.unregister(autofillId);
     widget.controller.removeListener(_didChangeTextEditingValue);
-    _floatingCursorResetController.dispose();
+    _floatingCursorResetController?.dispose();
```

Comment:
How do we get a dispose without an init state?

Reply:
also nit: null the variable after disposing it?

Reply:
> How do we get a dispose without an init state?

These weren't initialized in initState. Instead, the variable was marked as `late`, which initializes them on first access. If the floating cursor is never shown throughout the lifetime of this object (which is possible), then the first access was in dispose here.

Reply:
Can we add a test to make sure we don't regress this? Specifically a test that verifies we don't do whatever work these controllers would drive if we just immediately dispose the widget?

Reply:
Hm, I don't see a good way to test this. The test would have to verify that dispose doesn't instantiate any AnimationControllers.

Reply:
Ahh agreed, I misunderstood what's going on here. Filed https://github.com/dart-lang/linter/issues/3058 which could help with this.